plugins/woocommerce-admin/client/task-lists/fills/products/utils.tsx

```diff
@@ -2,31 +2,47 @@
  * External dependencies
  */
 import { intersection } from 'lodash';
+import { applyFilters } from '@wordpress/hooks';
 
 /**
  * Internal dependencies
  */
 import {
-	productTypes,
+	productTypes as baseProductTypes,
 	ProductType,
 	ProductTypeKey,
 	onboardingProductTypesToSurfaced,
 	supportedOnboardingProductTypes,
 	defaultSurfacedProductTypes,
+	SETUP_TASKLIST_PRODUCT_TYPES,
 } from './constants';
 
 export const getProductTypes = ( {
 	exclude,
 }: {
 	exclude?: ProductTypeKey[];
 } = {} ): ProductType[] => {
+	/**
+	 * Filter for adding custom product types to tasklist.
```

Comment:
Similar suggestion with [existing experimental filter](https://github.com/woocommerce/woocommerce/blob/2f77de3f185644c0ef3b9b9cb5cdae6f9ca7b31e/plugins/woocommerce-admin/client/analytics/report/variations/table.js#L295)

```suggestion
	 * Experimental: Filter for adding custom product types to tasklist.
```